libraries/WiFi/src/WiFiProv.cpp

```diff
@@ -0,0 +1,138 @@
+    License as published by the Free Software Foundation; either
+    version 2.1 of the License, or (at your option) any later version.
+  
+    This library is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+    Lesser General Public License for more details.
+  
+    You should have received a copy of the GNU Lesser General Public
+    License along with this library; if not, write to the Free Software
+    Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301  USA
+    
+*/
+
+#include <stdio.h>
+#include <stdint.h>
+#include <stdbool.h>
+#include <string.h>
+#include <esp_err.h>
+#include <esp_wifi.h>
+#include <esp_event_loop.h>
+#include <esp32-hal.h>
+
+#include <nvs_flash.h>
+#include <wifi_provisioning/scheme_ble.h>
+#include <wifi_provisioning/scheme_softap.h>
+#include <wifi_provisioning/manager.h>
+
+#include "WiFi.h"
+extern void tcpipInit();
+extern esp_err_t postToSysQueue(system_prov_event_t *);
+static const char *TAG = "Provision_in_Arduino";
+static bool prov_enable = false;
+uint8_t custom_service_uuid[16] = {  0xb4, 0xdf, 0x5a, 0x1c, 0x3f, 0x6b, 0xf4, 0xbf,
+                                     0xea, 0x4a, 0x82, 0x03, 0x04, 0x90, 0x1a, 0x02, };
+
+#define SERV_NAME_PREFIX_BLE "BLE_"
+#define SERV_NAME_PREFIX_WIFI "WIFI_"
+bool isProvEnabled(){
+    return prov_enable;
+}
+static void prov_event_handler(void *user_data, wifi_prov_cb_event_t event, void *event_data)
+{
+    if(event)
+    { 
+        system_prov_event_t *sys_prov = (system_prov_event_t *)malloc(sizeof(system_prov_event_t));
+        sys_prov->prov_event = (wifi_prov_event_t *)malloc(sizeof(wifi_prov_event_t));
+        sys_prov->sys_event = (system_event_t *)malloc(sizeof(system_event_t));
+        sys_prov->prov_event->event = event;
+        sys_prov->prov_event->event_data = event_data;
+        esp_err_t check = postToSysQueue(sys_prov);
+        if(check == ESP_FAIL)
+            log_e("Provisioning event not send to queue");
+    }
+}
+
+static void get_device_service_name(scheme prov_scheme, char *service_name, size_t max)
+{
+     uint8_t eth_mac[6];
+     WiFi.macAddress(eth_mac);
+     if(prov_scheme == WIFI_PROV_SCHEME_BLE)
+         snprintf(service_name, max, "%s%02X%02X%02X",SERV_NAME_PREFIX_BLE, eth_mac[3], eth_mac[4], eth_mac[5]);
+     else
+         snprintf(service_name, max, "%s%02X%02X%02X",SERV_NAME_PREFIX_WIFI, eth_mac[3], eth_mac[4], eth_mac[5]);  
+}
+
+void WiFiProvClass :: beginProvision(scheme prov_scheme, wifi_prov_security_t security, wifi_prov_event_handler_t scheme_event_handler,char * pop, char *service_name, char *service_key, uint8_t * uuid)
+{
+    prov_enable = true;
+    tcpipInit();
+    bool provisioned = false;
+    wifi_init_config_t cfg = WIFI_INIT_CONFIG_DEFAULT();
+    ESP_ERROR_CHECK(esp_wifi_init(&cfg));
```

Comment:
this is the IDF way :) please check if returned `ESP_OK` and if not error and exit.

Reply:
YES